setup.py

```diff
@@ -793,16 +793,42 @@ def run(self):
         import subprocess
         import sys
         import warnings
+        import pathlib
 
         def check_linter_exists(linter):
             if shutil.which(linter) is None:
                 msg = "Please install '%s' in your environment. (hint: 'python3 -m pip install %s')"
                 warnings.warn(msg % (linter, linter))
                 sys.exit(1)
 
-        c_files_unfiltered = glob.glob("src_c/**/*.[ch]", recursive=True)
-        c_file_disallow = ["_sdl2", "pypm", "SDL_gfx", "sse2neon.h", "src_c/doc/", "_sprite.c"]
-        c_files = [x for x in c_files_unfiltered if not any([d for d in c_file_disallow if d in x])]
+        def filter_files(path_obj, all_files, allowed_files, disallowed_files):
+            files = []
+            for file in all_files:
+                for disallowed in disallowed_files:
+                    if file.match(str(path_obj / disallowed)):
+                        break
+                else:  # no-break
+                    files.append(str(file))
+
+                for allowed in allowed_files:
+                    if file.match(str(path_obj / allowed)):
+                        files.append(str(file))
+                        break
```

Comment:
Since the allowed files must first be disallowed for this function to work, I think it should be possible to combine the if statements and reduce the code to one inner for loop:

```
for disallowed in disallowed_files:
    if file.match(str(path_obj / disallowed)) and not file.match(str(path_obj / allowed)):
        break
else:  # no-break
    files.append(str(file))
```

Or something like that. With the current design you could theoretically have a file that was in allowed, but not in disallowed added to the returned files list twice.

Reply:
This has been fixed now by the looks of things, an allowed file won't be added twice if it's not disallowed. However the loop simplification you suggested may not work here because there are two different lists (allowed/disallowed)